packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -221,10 +221,9 @@ class CupertinoSheetTransition extends StatefulWidget {
       reverseCurve: reverseCurve,
       parent: secondaryAnimation,
     );
-    final double deviceCornerRadius = MediaQuery.maybeViewPaddingOf(context)?.top ?? 0;
 
     final Animatable<BorderRadiusGeometry> decorationTween = Tween<BorderRadiusGeometry>(
-      begin: BorderRadius.circular(deviceCornerRadius),
+      begin: BorderRadius.circular(0),
```

Comment:
``` dart
begin: BorderRadius.vertical(
  top: Radius.circular(roundedDeviceCorners ? deviceCornerRadius : 0)
),
```
So we can animate from zero for flat screens and use the smoother transition for rounded corners. We also should only apply the radius to the top of the view. The bottom is unnecessary.